syntax = "proto3";

option go_package = "tiaoyuyu.com/pb/gamepb";
option java_multiple_files = true;
option java_package = "gamepb";
option java_outer_classname = "gamepb";

package gamepb;

import "error.proto";


// The greeting service definition.
service Greeter {
    // Sends a greeting
    rpc SayHello (HelloRequest) returns (HelloReply) {}
    rpc RouteChat(stream RouteNote) returns (stream RouteNote) {}

}

// The request message containing the user's name.
message HelloRequest {
    string name = 1;
}

// The response message containing the greetings
message HelloReply {
    string message = 1;
}
message RouteNote {
    Point location = 1;
    string  message = 2;
}

//============DATA===============//

message Error {
    ErrorType errorCode = 1;
    string errorMsg = 2;
}

// 位置 及 阵营
message Point {
    int32 x = 1;
    int32 y = 2;
    int32 Camp = 3;
}

// 位置 阵营 及 次序
message ChessStep {
    Point point = 1; // 位置 及 阵营
    int32 index = 2; // 当前局的第几步
}

// 当前对局
message GobangInfo {
    repeated ChessStep chessSteps = 1;
}
//===============================//

//============PROTOCOL===========//

// 走一步
message C2SGobangStep {
    Point point = 1;
}

// 当前局势
message S2CGobangStep {
    Error error = 1;
    GobangInfo gobangInfo = 2;
}

message C2SPing {
    int64 timestamp = 1;
}
message S2CPing {
    Error error = 1;
    int64 timestamp = 2;
}
//==========================//